draft-ietf-gnap-core-protocol.md

```diff
@@ -953,6 +1044,9 @@ string-type resource items.
     ]
 ~~~
 
+The requested access is the sum of all elements of the array, including both objects and 
+reference strings.
+
```

Comment:
I'm not sure if this belongs here or should be a separate issue: we should be clear on the semantics of the request/response.
* If the response contains a `resources` object, the RC is granted **exactly** this access. 
* If it doesn't, the RC is granted **exactly** the access that was requested.

The word "exactly" here means that anything outside this access (other actions, other locations, or any unspecified combination) MUST be rejected by the RC.

Reply:
I think this is a separate issue. I'm personally more in favor of the response always containing the `resources` or equivalent for every access token. OAuth 2 has a similar optimization with `scope` where you don't have to return it if you don't want to, but I think it might make more sense to just always return it and the client can ignore it if it doesn't need anything from it.

Reply:
I think Yaron's question is interesting. It may well be that the request is very different from the actual access that is provided. We might indeed always return a resource section, which could potentially just return "same as requested" if there was no limitation. Or we could always return it in full. 
But in practice what matters is the access token itself, which is really the source of truth. Do we plan to implement token introspection (and is it mandatory or not)?

-> probably a separate issue.

Reply:
Introspection is a service for RS's and not clients (see #99), and so an introspection response might be different from the access token response to the client. 

I've opened @yaronf 's comment in a new issue: #141.

Reply:
If not called introspection, wasn't there something similar (maybe that was called management Uri if I recall correctly?)

Reply:
I'm marking this conversation resolved so that it can continue in the new issue.